tests/parser.cpp

```diff
@@ -2747,6 +2747,115 @@ BPFTRACE_STACK_MODE=perf; i:s:1 { exit(); }
 )");
 }
 
+TEST(Parser, opts)
+{
+  test("opts = [{ blah = 5 }] BEGIN {}", R"(
+Program
+ opts
+  opt
+   =
+    var: blah
+    int: 5
+ BEGIN
+)");
+
+  test("opts = [{ blah = 5; type = string }] BEGIN {}", R"(
```

Comment:
Thinking out loud. I'm not sure I understand the general syntax. Why not just have a list, e.g.

```
opts = {
   foo = "bar"; // type is a string, default is "bar"
   zap = 5; // type is an integer, default is 5
}
```

You could special case `foo = 0` and accept `--foo` to set to 1 (and `foo = 1` to accept `--no-foo`, if you wanted). Nothing prevents extending beyond this, but I'd just be curious to understand more advanced user examples.

Reply:
Got it. I thought about this today, and put a bit of a straw man there as a simplifying alternative.

My 2c is that it would be good to have some concrete examples, look at existing scripts a bit, and maybe let this stack up a bit to land. I don't object to the change, I just think it's worthwhile to be specifically cautious for new syntax.